File: src/main/java/me/untouchedodin0/privatemines/mine/MineType.java
package me.untouchedodin0.privatemines.mine;

import org.bukkit.Material;
import redempt.redlib.config.annotations.ConfigMappable;
import redempt.redlib.config.annotations.ConfigPath;

import java.util.Map;

@ConfigMappable
public class MineType {

    @ConfigPath
    private String name;
    private String file;
    private int resetTime;
    private double resetPercentage;
    private Map<Material, Double> materials;

    public String getName() {
        return name;
    }

    public String getFile() {
        return file;
    }

    public int getResetTime() {
        return resetTime;
    }

    public Map<Material, Double> getMaterials() {
        return materials;
    }
}

File: src/main/kotlin/me/untouchedodin0/kotlin/MineType.kt
package me.untouchedodin0.kotlin

import me.untouchedodin0.privatemines.configmanager.annotations.ConfigMappable
import me.untouchedodin0.privatemines.configmanager.annotations.ConfigPath
import org.bukkit.Material

@ConfigMappable
class MineType {
    @ConfigPath
    val name: String? = null
    val file: String? = null
    val resetTime = 0
    val resetPercentage = 0.0
    val materials: Map<Material, Double>? = null
}
